src/honeybot/api/main.py

```diff
@@ -25,6 +25,162 @@
+        irc (socket.socket): The socket object for the IRC connection.
+        is_listen_on (int): Flag indicating whether the bot is listening for incoming messages.
+        domain (str): The domain of the server URL.
+        sp_command (str): The special command for the bot.
+        plugins (list): A list of loaded plugins.
+        core_plugins (list): A list of core plugins.
+
+    Methods:
+        message_info(s): Parses a message string and returns a dictionary containing information about the message.
+        bot_info(): Returns a dictionary containing information about the bot.
+        methods(): Returns a dictionary of methods used by the bot.
+        send(msg): Sends a raw message to the server.
+        send_target(target, msg): Sends a message to a specific target (channel or user).
+        join(channel): Joins a channel.
+        is_valid_plug_name(name): Checks if a plugin name is valid.
+        print_running_infos(): Prints the running information of the bot.
+        load_plugins_from_folder(category_folder, from_conf, from_dir): Loads plugins from a specific folder.
+        load_plugins(): Loads the plugins specified in the configuration.
+        run_plugins(incoming): Runs the loaded plugins for an incoming message.
+        core_commands_parse(incoming): Parses the core commands for an incoming message.
+        connect(): Connects to the server.
+        identify(): Sends the identification command to the server.
+        greet(): Sends the initial commands to the server for bot registration.
+        pull(): Listens for incoming messages from the server.
+        quit(): Sends the quit command to the server.
+        stay_alive(incoming): Handles the stay alive functionality for the bot.
+        registered_run(): Runs the bot as a registered bot.
+        unregistered_run(): Runs the bot as an unregistered bot.
+    """
+
+    def __init__(self, info, password=""):
+        self.info = info
+        with open(info["toml_path"], "rb") as f:
+            self.configs = tomli.load(f)
+        self.settings_path = self.info["settings_path"]
+        self.root_path = self.info["cwd"]
+        self.server_url = self.configs["INFO"]["server_url"]
+        self.port = int(self.configs["INFO"]["port"])
+        self.name = self.configs["INFO"]["name"]
+        self.owners = self.configs["USERNAMES"]["owners"]
+        self.password = password
+        self.friends = self.configs["USERNAMES"]["friends"]
+        self.autojoin_channels = self.configs["INFO"]["autojoin_channels"]
+        self.downloaded_plugins_to_load = self.configs["PLUGINS"]["downloaded"]
+        self.required_modules = get_requirements()
+        self.time = time.time()
+
+        self.irc = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.is_listen_on = 1
+        dom = self.server_url.split(".")
+        self.domain = ".".join(dom[-2:])
+        self.sp_command = "hbot"
+        self.plugins = []
+        self.core_plugins = []
```

Comment:
`class BotCore` and its `__init__()` are duplicated 3 times, please remove the duplicates

First duplicate